service/lib/agama/registration.rb

```diff
@@ -26,9 +32,26 @@ class Registration
     # initializes registration with instance of software manager for query about products
     def initialize(software_manager)
       @software = software_manager
+      @on_state_change_callbacks = []
     end
 
-    def register(code, email: nil)
+    def register(code, email: "")
+      target_distro = "ALP-Dolomite-1-x86_64" # TODO read it
+      registration = Registration::Registration.new # intentional no url yet
```

Comment:
I think this would not work if `#announce_system` fails because `#catch_registration_errors` could call to YaST UI.

I would directly call to `#announce_system` and `#create_credentials_file` instead of using `Registration`.